provider_sources/_compress_providers.py

```diff
@@ -49,7 +49,18 @@
         )
 
 # combine both
-leaflet.update(xyz)
+
+for key, val in xyz.items():
+    if key in leaflet:
+        if any(
+            isinstance(i, dict) for i in leaflet[key].values()
+        ):  # for related group of bunch
```

Comment:
```suggestion
        if isinstance(val, Bunch): 
```

Is this what you're trying to get here? I think we can easily import `xyzservices.Bunch` above here and check directly. We'll just need to make sure that we have xyzservices install in our GHA here https://github.com/geopandas/xyzservices/blob/main/.github/workflows/update_providers.yaml (I can take care of that later).

Reply:
Yes, done